core/tests/webdriverio_utils/waitFor.js

```diff
@@ -194,7 +194,7 @@ var urlRedirection = async function(url) {
 var numberOfElementsToBe = async function(
     elementSelector, elementName, number, childSelector = null) {
   await browser.waitUntil(async function() {
-    var element = childSelector ? await $(elementSelector).$$(
+    var element = childSelector ? await elementSelector.$$(
```

Comment:
Element selector will be a classname only in the case when we will have childSelector because at some places we do not have access to the parent class selector and hence we were unable to use this function. So I modified this function so that this function can work in those cases as well.

Reply:
Sorry, I am not able to understand this? Can you please explain this in a more verbose way?

Reply:
You earlier said that we do not have access to parent class selector, so can we not wait for it's presence?

Reply:
Take a example like this: 
```
var selector = $('.class-name');
await form.exampleFunction(element);
```
In form.js we have
```
exampleFunction(elem) {
var element2 = await elem.$$('.second-class');
}
```
Now if we want to wait till the number of element2 to be n, then we need to pass the classname of both parent (elem) and children (.second-class) but as we are able to see we do not have access to elem classname (elem is element obtained by calling $ on class name, its not a class name) we need to pass the elem itself with the child class. So in `numberOfElementsToBe` i just replaced $(elementSelector) with elem.

Hope this will make things clear.